### PR TITLE
T4801: Re-add support for AWS with build-flavors

### PR DESCRIPTION
**Pull request by ssasso**
## Change Summary

On the old build script, it was possible to invoke
```
./configure
make AWS
```
With the new build system, a new **build-flavors** file needs to be created for supporting the AWS use case.

## Types of changes

 
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe):

Improvement/migration

## Related Task(s)

https://phabricator.vyos.net/T4801

## How to test

`./build-vyos-image <xxx> aws`

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by c-po — APPROVED** [review]

File: data/build-flavors/aws.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Generic (aka "universal") ISO image + AWS STUFF (to be able to create an AMI)
+
+image_format = "iso"
+
+packages = [
+  "qemu-guest-agent",
```
Comment (zdc) [inline data/build-flavors/aws.toml:6]: Is `qemu-guest-agent` used by AWS EC2?

File: data/build-flavors/aws.toml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  "cloud-init",
+  "awscli",
+]
+
+[[includes_chroot]]
+  path = "etc/cloud/cloud.cfg.d/90_dpkg.cfg"
+  data = '''
+datasource_list: [ Ec2 ]
```
Comment (zdc) [inline data/build-flavors/aws.toml]: It is better to add `None` in the end on the list.

File: data/build-flavors/aws.toml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            facility all {
+                level info
+            }
+            facility protocols {
+                level debug
+            }
+        }
+    }
+    ntp {
+        server "169.254.169.123"
```
Comment (zdc) [inline data/build-flavors/aws.toml:42]: I agree that the AWS server can be added to the list, but the original ones should be kept as well.

File: data/build-flavors/aws.toml (hunk 1 above)
Comment (ssasso) [reply data/build-flavors/aws.toml:6]: AWS EC2 is kvm under the hood.

File: data/build-flavors/aws.toml (hunk 3 above)
Comment (ssasso) [reply data/build-flavors/aws.toml:42]: An instance deployed on AWS may not have access to external ones - while AWS' one is always available, and AWS itself suggests the use of that one on EC2 instances.

File: data/build-flavors/aws.toml (hunk 2 above)
Comment (ssasso) [reply data/build-flavors/aws.toml]: done - even though the original one does not have it: https://github.com/vyos/vyos-build/blob/current/tools/cloud-init/AWS/90_dpkg.cfg

**Comment by sempervictus** [comment]
If this sort of thing is of interest, my  (revised & salvaged) `openstack` target runs perfectly fine on AWS with `cloud-init` as well as being able to use other clouds' metadata modalities. We have a packer process for installing the cloud-targeted ISO to a small disk image and another for uploading the results to AWS using Packer's EBS surrogate process.

**Comment by andamasov** [comment]
@dmbaturin can you check this

**Comment by dmbaturin** [comment]
Additional flavors are soon to be moved to a separate repository.

**Comment by sempervictus** [comment]
Thanks for the heads up. Is there a roadmap for this in phabricator somewhere? Been hacking on repo reactively to stay up w changes, might have to do less of that if I track what's coming :)

**Pull request closed without merging**